File: Algo/Medium/0647_Palindromic_Substrings/Solution.py
from typing import *

## 2022-05-21
class Solution:
    def countSubstrings(self, s: str) -> int:
        def howManyPalindromes(left, right):
            count = 0
            while (left >= 0 and right < len(s) and s[left] == s[right]):
                count += 1
                left -= 1
                right += 1
            return count
        
        palindrome_count = 0
        for i in range(len(s)):
            palindrome_count += howManyPalindromes(i, i)
            if i > 0:
                palindrome_count += howManyPalindromes(i-1, i)
        
        return palindrome_count

File: Algo/Medium/0647_Palindromic_Substrings/Solution.cpp
// 2022-05-22
class Solution {
public:
    int howManySubstrings(string &s, int left, int right) {
        int substrings = 0;
        while (left >= 0 && right < s.length() && s[left] == s[right]) {
            left--; 
            right++;
            substrings++;
        }
        return substrings;
    }
    
    int countSubstrings(string s) {
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            count += howManySubstrings(s, i, i);
            if (i != 0)
                count += howManySubstrings(s, i-1, i);
        }
        return count;
    }
};